### PR TITLE
New Starboard-Python support

### PR DESCRIPTION
**Pull request by stefnotch**
Changes required to make the [new starboard-python](https://github.com/gzuidhof/starboard-python/pull/5) work with starboard-notebook.

We should also document the COOP/COEP requirements somewhere.

### CODE REVIEW

File: src/initServiceWorker.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    }
+  );
+
+  navigator.serviceWorker.getRegistration().then(function (reg) {
+    // TODO: Or use some other solution
+    // https://stackoverflow.com/questions/51597231/register-service-worker-after-hard-refresh
+    // https://github.com/mswjs/msw/issues/98#issuecomment-612118211
+    if (reg !== undefined && reg.active && !navigator.serviceWorker.controller) {
+      window.location.reload();
+    }
```
Comment (stefnotch) [inline src/initServiceWorker.ts:17]: This doesn't seem to work, as in, the site doesn't get reloaded. I should probably use one of the other approaches anyways

File: src/main.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 import { StarboardNotebookElement } from "./components/notebook";
+import "./initServiceWorker";
```
Comment (stefnotch) [inline src/main.ts:6]: This should be optional! We don't want to forcibly register a service worker for every user. I'm not sure how to best do that though.

File: src/sw.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+self.addEventListener("activate", (event) => {
+  event.waitUntil(self.clients.claim());
+});
+
+self.addEventListener("fetch", function (event) {
+  event.respondWith(
+    fetch(event.request)
+      .then(function (response) {
+        // We only need to set the headers for index.html
+        if (!response.url.includes("index.html")) return response;
```
Comment (stefnotch) [inline src/sw.js:14]: At least it appears that we only need the headers for `index.html`. I'm not entirely sure if that's 100% spec compliant though, maybe we should set them for everything.

**Review by stefnotch — COMMENTED** [review]
Interestingly enough, running this on a non-localhost site and forcibly enabling service workers in Firefox by setting `devtools.serviceWorkers.testing.enabled` completely breaks.

File: src/main.ts (hunk 2 above)
Comment (gzuidhof) [reply src/main.ts:6]: I'm not 100% sure either.. One could put it in the search query, but `unpkg` CDN removes any querystring which makes ad-hoc embedding more difficult (and other CDNs generally don't serve html files with the correct content type header (something like `text/html`)).

We could maybe ship two different `index.html` files, we can ship an additional one that registers the code in `initServiceWorker` prior to loading the nb at all. Still not great.. But I guess many notebooks won't need [coop/coep](http://www.foddy.net/Athletics.html)? 

We can also put it in the `runtimeConfig`, perhaps that's the cleanest approach (although it does reload rather late then.. but I guess that's a one-time pain)

Another option (sorry for the overload): we expose it under `runtime.exports.core.register<some_name>HeaderServiceWorker`, which takes argument `{autoReload: boolean}`, and returns something like `{reloadRequired: boolean}`. Then within a notebook itself you can enable it, and if you want to get fancy you can warn the user what is about to happen (and they can save their work or something).

File: src/initServiceWorker.ts (hunk 1 above)
Comment (stefnotch) [reply src/initServiceWorker.ts:17]: Seems to work now

File: src/main.ts (hunk 2 above)
Comment (stefnotch) [reply src/main.ts]: I decided to put it in `runtimeConfig`

File: src/sw.js (hunk 3 above)
Comment (stefnotch) [reply src/sw.js]: I decided to comment this bit out

**Review by gzuidhof — APPROVED** [review]
Looks great :) thank you

**Comment by gzuidhof** [comment]
Just one request: could you add a summary of the changes to the changelog? :)

**Comment by stefnotch** [comment]
> 
> 
> Just one request: could you add a summary of the changes to the changelog? :)

Okay, I added it to the changelog :) Sorry for the merge conflicts though

**Pull request merged**